### PR TITLE
Solution by Mario Bisignani

### PR DESCRIPTION
**Pull request by bisignam**
### Short Description
Hello, 3AP-AG this is the assignment solution by Mario Bisignani. 

Regarding the part 5 of the assignment, I found a couple of "interesting" things:

 - In Switzerland, there are 1161 postal communities associated to more than one political community

- The municipality of Urnäsch is associated to 7 political communities:  Hemberg,Nesslau,Wildhaus-Alt St. Johann,Schwende,Urnäsch,Schönengrund and Hundwil

Thanks for your time

Have a nice weekend.

Mario

### CODE REVIEW

File: src/main/java/ch/aaap/assignment/model/Canton.java [hunk 1]
```diff
@@ -1,10 +1,23 @@
 package ch.aaap.assignment.model;
 
-public interface Canton {
+import java.util.Set;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/model/Canton.java:8]: unused import

File: src/main/java/ch/aaap/assignment/model/District.java [hunk 2]
```diff
@@ -1,10 +1,21 @@
 package ch.aaap.assignment.model;
 
-public interface District {
+import java.util.Set;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/model/District.java:8]: unused import

File: src/main/java/ch/aaap/assignment/model/Canton.java [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
+import lombok.experimental.FieldDefaults;
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/model/Canton.java:9]: I personally would not use experimental api's when it is not really required. but this is opinionated 😄

File: src/main/java/ch/aaap/assignment/model/PoliticalCommunity.java [hunk 4]
```diff
@@ -1,16 +1,40 @@
...
-  public String getName();
-
-  public String getShortName();
-
-  public LocalDate getLastUpdate();
-
-  // TODO add more features here representing the relations
+  @Override
+  public String toString() {
+    return "PoliticalCommunity{" +
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/model/PoliticalCommunity.java:28]: I like, makes toString more readable compared to the default

File: src/main/java/ch/aaap/assignment/Application.java [hunk 5]
```diff
@@ -1,11 +1,21 @@
...
 import ch.aaap.assignment.raw.CSVUtil;
 import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class Application {
```
Comment (robertblust) [inline src/main/java/ch/aaap/assignment/Application.java:20]: Good pragmatic implementation 👍 
- well documented
- nice usage of java streams features

File: src/main/java/ch/aaap/assignment/model/Canton.java (hunk 3 above)
Comment (bisignam) [reply src/main/java/ch/aaap/assignment/model/Canton.java:9]: Thanks for the feedback, I agree,  I find @FieldDefaults so clean from time to time I forget it is an experimental feature. I hope they will promote it to a standard feature soon.

File: src/main/java/ch/aaap/assignment/model/District.java (hunk 2 above)
Comment (bisignam) [reply src/main/java/ch/aaap/assignment/model/District.java:8]: argh 🍡 my fault. I forgot to reorganize imports.

**Comment by bisignam** [comment]
Thanks a lot for your feedback. Hope to hear from you soon.

**Pull request closed without merging**